### ISSUE TITLE
Style/HashAsLastArrayItem for array of hashes

### ISSUE THREAD

**Issue by dvandersluis** [rubocop-hq/rubocop#8617]
I believe this might be similar to #8409, but the reverse. When `EnforcedStyle` is `no_braces`, `Style/HashAsLastArrayItem` is a bit over-zealous IMO when it comes to an array of *only* hashes, and suggests removing the outer braces for the last element, which is going to lead to confusion.

Can the cop be updated to not consider arrays that are only hashes?

--------

## Expected behavior

Cop should not find any offense with this code with `EnforcedStyle: no_braces`:

```ruby
QUOTES = [
  {
    background_color: '#EE0F0F',
    id: 2987584,
    quote_author_id: 8202,
    quote_author_title: 'John Smith',
    image_filename: 'image-2987584.png'
  },
  {
    background_color: '#EE0F0F',
    id: 2948124,
    quote_author_id: 1709304,
    quote_author_title: 'Tim Brown',
    image_filename: 'image-2948124.png'
  }
].freeze
```

## Actual behavior

The cop registers offenses for the outer braces on the last element, and autocorrects them away:

```ruby
QUOTES = [
  {
    background_color: '#EE0F0F',
    id: 2987584,
    quote_author_id: 8202,
    quote_author_title: 'John Smith',
    image_filename: 'image-2987584.png'
  },

  background_color: '#EE0F0F',
  id: 2948124,
  quote_author_id: 1709304,
  quote_author_title: 'Tim Brown',
  image_filename: 'image-2948124.png'
].freeze
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
0.89.1 (using Parser 2.7.1.4, rubocop-ast 0.3.0, running on ruby 2.6.2 x86_64-darwin18)
```
